lib/iex/cloud/request.rb

```diff
@@ -20,16 +39,17 @@ def delete(path, options = {})
       private
 
       def request(method, path, options)
-        path = [endpoint, path].join('/')
+        options = options.dup
+        path = [options.delete(:endpoint) || endpoint, path].join('/')
         response = connection.send(method) do |request|
+          request.options.merge!(options.delete(:request)) if options.key?(:request)
```

Comment:
Explain this change? Shouldn't be needed.

Reply:
if we leave anything extra in the options hash then in line 48 (`request.url(path, options)`) all those garbage will be added to the request URL as parameters

Reply:
But I think that's the intent, if you pass something in it should land in request params, but I could be wrong. Ayway, is this related to streaming? What's broken without this change?

Reply:
I pass `request: { on_data: event_parser }` as additional option to parse the response in realtime. If I don't delete it then it adds like `request%5Bon_data%5D=%23%3CProc%3A0x00007feea6c5e670%40%28pry%29%3A3%3E` in the URL. Not critical, IEX just ignores it but any way, I think we should filter out all config options like `request`, `endpoint` etc.

Reply:
Got it, agreed.